lib/hyacinth/adapters/digital_object_search_adapter/solr.rb

```diff
@@ -55,6 +63,20 @@ def solr_params_for(search_params)
           solr_parameters
         end
 
+        # Adds filter queries to the given solr_prameters based on projects where the
+        # given user has read access.
+        def apply_user_project_filters(solr_parameters, user)
+          read_objects_access_project_string_keys = Project.accessible_by(Ability.new(user), :read_objects).map(&:string_key)
+          if read_objects_access_project_string_keys.present?
+            solr_parameters.fq('projects_ssim', read_objects_access_project_string_keys)
+          else
+            # User has no project permissions, so we'll deliberately put in an unresolvable value
+            # to get zero results. Projects cannot have exclamation points in their string keys,
+            # so the value below will work.
+            solr_parameters.fq('projects_ssim', '!nomatch!')
```

Comment:
Bailing out early would be better, yeah.  The above implementation is similar to how we do it in Hyacinth 2, but it would be nice not to waste a search with a filter that deliberately doesn't match anything.

One unfortunate thing about the current search implementation is that it always returns a solr response (rather than something like a custom HyacinthSearchResultSet), so if we do decide to return an empty set that means faking an entire empty solr response.

How would you feel if we ticket your suggestion and refactor at a later time once we change the search adapter result type?  That way we won't be feeing more into the solr-response-as-search pattern.

Reply:
I thought maybe there was a RSolr response object we could use to return an empty response, but I might be thinking about Blacklight.

Ticketing this for later work sounds good, i'll create the ticket.

Reply:
HYACINTH-555

Reply:
Thanks!